src/common/vitals/first-interaction.js

```diff
@@ -1,24 +1,34 @@
 import { VitalMetric } from './vital-metric'
-import { VITAL_NAMES } from './constants'
+import { VITAL_NAMES, PERFORMANCE_ENTRY_TYPE } from './constants'
 import { initiallyHidden, isBrowserScope } from '../constants/runtime'
 
 // Note: First Interaction is a legacy NR timing event, not an actual CWV metric
-// ('fi' used to be detected via FID.  It is now represented by the first INP)
 export const firstInteraction = new VitalMetric(VITAL_NAMES.FIRST_INTERACTION)
 
-export function recordFirstInteraction (attribution) {
-  if (isBrowserScope) {
+if (isBrowserScope) {
+  try {
+    let observer
     // preserve the original behavior where FID is not reported if the page is hidden before the first interaction
-    if (initiallyHidden || firstInteraction.isValid) return
-    const attrs = {
-      type: attribution.interactionType,
-      eventTarget: attribution.interactionTarget,
-      loadState: attribution.loadState
-    }
+    if (PerformanceObserver.supportedEntryTypes.includes(PERFORMANCE_ENTRY_TYPE.FIRST_INPUT) && !initiallyHidden) {
+      observer = new PerformanceObserver((list) => {
+        const firstInput = list.getEntries()[0]
+
+        const attrs = {
+          type: firstInput.name,
+          eventTarget: firstInput.target
+        }
 
-    firstInteraction.update({
-      value: attribution.interactionTime,
-      attrs
-    })
+        observer.disconnect()
+        if (!firstInteraction.isValid) {
+          firstInteraction.update({
+            value: firstInput.startTime,
+            attrs
+          })
+        }
+      })
+      observer.observe({ type: PERFORMANCE_ENTRY_TYPE.FIRST_INPUT })
```

Comment:
I think this probably needs to include the buffered argument to capture cases where an interaction has occurred before this module was executed